frontend/src/store/task.store.ts

```diff
@@ -0,0 +1,44 @@
+import { create } from 'zustand';
+import { Task } from '../types/Task';
+
+type TaskStoreType = {
+  task: Task;
+  setTask: (task: Task) => void;
+};
+
+type CompletedStoreType = {
+  completed: Record<number | string, boolean>; // Task ID to completion status mapping
+  setCompleted: (taskId: string, completed: boolean) => void;
+};
+
+export const initialValue = {
+  id: '',
+  title: '',
+  description: '',
+  status: '',
+  due_date: '',
+  assignee: '',
+  completed: false,
+  priority: '',
+  project_id: 0,
+  user_id: '',
+};
```

Comment:
Kristina added functionality to add new tasks, but the default values here don't match the ones she have made. Maybe you guys wanna sync on this :)